Input/GamepadTest/source/Gamepad.hx

```diff
@@ -2,195 +2,274 @@ package;
 
-		crosshairs = createSprite(0, 0, "crosshairs", 1);
+		crosshairs = createSprite(0, 0, "crosshairs");
 		crosshairs.visible = false;
+		
+		updateInputSprites();
+		add(inputLabels);
+		inputLabels.x = 0;
+		inputLabels.y = 0;
 	}
 
-	function createSprite(x:Float, y:Float, ?fileName:String, alpha:Float = -1):FlxSprite
+	function createSprite(x:Float, y:Float, ?fileName:String):FlxSprite
 	{
-		if (alpha == -1)
-			alpha = ALPHA_OFF;
 		if (fileName != null)
 			fileName = 'assets/$fileName.png';
 
 		var sprite = new FlxSprite(x, y, fileName);
-		sprite.alpha = alpha;
 		sprite.antialiasing = true;
-		return add(sprite);
+		sprite.offset.set(sprite.width / 2, sprite.height / 2);
+		sprite.x += sprite.offset.x;
+		sprite.y += sprite.offset.y;
+		add(sprite);
+		return sprite;
+	}
+	
+	function createInputSprite(x:Float, y:Float, ?fileName:String, input:InputID, type:InputType = Digital, offset:FlxVector = null):FlxSprite
+	{
+		var sprite = createSprite(x, y, fileName);
+		var label = createLabel(sprite.x, sprite.y, "", 16);
+		label.borderColor = LABEL_OFF;
+		if (offset != null)
+		{
+			inputLabels.add(drawLine(label.x, label.y, offset, LABEL_OFF));
+			label.x += offset.x;
+			label.y += offset.y;
+			offset.putWeak();
+		}
+		inputLabels.add(label);
+		
+		switch(type)
+		{
+			case Digital: sprite.alpha = ALPHA_OFF;
+			case Invisible: sprite.visible = false;
+			default:
+		}
+		
+		inputSprites.push({ sprite:sprite, input:input, type:type, label:label });
+		return sprite;
 	}
 
-	function createBar(x:Float, y:Float):FlxBar
+	function createBar(x:Float, y:Float, label:String):Bar
 	{
 		var bar = new FlxBar(x, y, FlxBarFillDirection.LEFT_TO_RIGHT, 100, 14, null, "", 0, 100, true);
 		bar.createGradientBar([FlxColor.BLACK], [FlxColor.RED, FlxColor.BLUE], 1, 180, true, FlxColor.BLACK);
 		add(bar);
-		return bar;
+		var label = createLabel(x, y, label);
+		add(label);
+		return { bar:bar, label:label };
 	}
 
-	function createLabel(x:Float, y:Float, ?label:String):FlxText
+	function createLabel(x:Float, y:Float, label:String, size:Int = 8):FlxText
 	{
-		var label = new FlxText(x, y, 0, label);
+		var label = new FlxText(x, y, 0, label, size);
 		label.color = FlxColor.WHITE;
-		label.setBorderStyle(FlxTextBorderStyle.OUTLINE_FAST, FlxColor.BLACK, 1, 1);
-		add(label);
+		label.setBorderStyle(FlxTextBorderStyle.OUTLINE, FlxColor.BLACK, size/8, size/8);
+		label.autoSize = true;
 		return label;
 	}
-
+	
+	function drawLine(x:Float, y:Float, offset:FlxVector, color:FlxColor):FlxSprite
+	{
+		var line = new FlxSprite(x, y);
+		line.makeGraphic(Math.floor(offset.length), 2, color);
+		line.angle = offset.degrees;
+		line.origin.set(1, 1);
+		return line;
+	}
+	
+	inline function createStick(x:Float, y:Float, input:InputID)
+	{
+		var isLeft = input == LEFT_ANALOG_STICK;
+		var stick = createInputSprite(x, y, "Stick", input, Stick);
+		x = stick.x - this.x;// prevent double nested offset from spritegroup
```

Comment:
If hacks like this are needed, should it really be a `FlxSpriteGroup`?

Reply:
making it a FlxGroup would mean adding a different hack IMO. I will look into that option, though